src/datepicker/datepicker.js

```diff
@@ -605,6 +611,9 @@ function(scope, element, attrs, $compile, $parse, $document, $rootScope, $positi
     datepickerEl = angular.element(popupEl.children()[0]);
     datepickerEl.attr('template-url', datepickerTemplateUrl);
 
+    scope.ngModelOptions = { $options: angular.extend({ allowInvalid: false }, ngModelOptions) };
```

Comment:
I don't think it's necessary, I copied some of the code from @davious (forgot to mention that originally in the pull request) and he had it in there.